packages/app/src/specs/RowDirectory.vue

```diff
@@ -4,37 +4,75 @@
     :data-cy="`row-directory-depth-${depth}`"
     :aria-expanded="expanded"
   >
-    <i-cy-chevron-down-small_x16
+    <svg
+      width="16"
+      height="16"
+      viewBox="0 0 16 16"
+      fill="none"
+      xmlns="http://www.w3.org/2000/svg"
       class="
         mr-8px text-sm icon-dark-gray-300
         group-hocus:(icon-dark-gray-700)
       "
-      :class="{'transform rotate-270': !expanded}"
-    />
-    <component
-      :is="IconFolder"
+    >
+      <path
+        d="M4 6L8 10L12 6"
+        stroke="currentColor"
+        class="icon-dark"
+        stroke-width="2"
+        stroke-linecap="round"
+        stroke-linejoin="round"
+      />
+    </svg>
+    <svg
+      width="16"
+      height="14"
+      viewBox="0 0 16 14"
+      fill="none"
+      xmlns="http://www.w3.org/2000/svg"
       class="icon-dark-white icon-light-gray-200"
-    />
+    >
+      <path
+        d="M14 13C14.5523 13 15 12.5523 15 12V4C15 3.44772 14.5523 3 14 3H9L7.29289 4.70711C7.10536 4.89464 6.851 5 6.58579 5H1V12C1 12.5523 1.44772 13 2 13H14Z"
+        fill="#2E3247"
+        class="icon-dark"
+      />
+      <path
+        d="M9 3L7.29289 1.29289C7.10536 1.10536 6.851 1 6.58579 1H2C1.44772 1 1 1.44772 1 2V5M9 3H14C14.5523 3 15 3.44772 15 4V12C15 12.5523 14.5523 13 14 13H2C1.44772 13 1 12.5523 1 12V5M9 3L7.29289 4.70711C7.10536 4.89464 6.851 5 6.58579 5H1"
+        stroke="#434861"
+        stroke-width="2"
+        stroke-linejoin="round"
+        class="icon-light"
+      />
+    </svg>
     <div
       :title="props.name"
       class="text-gray-600 truncate"
     >
       <HighlightedText
+        v-if="props.highlightText"
         :text="props.name"
         :indexes="props.indexes"
         class="font-medium"
         highlight-classes="text-gray-1000"
       />
+      <span v-else>
```

Comment:
This seems like the real problem, not the SVG/icons. Can you estimate the impact of these two optimizations?

Also, if we are making two optimizations, would 2 PRs be better? I'm almost sure that one of these is doing the majority of the performance improvements, and I think the SVG change is going to be a massive headache in the future (inline SVG is basically write-only, no way to update/change/associate with a design system).

Reply:
Hey, thanks for reviewing! Yes let's sync (maybe at the start of your day tomorrow).

The `<HighlightText>` changes weren't that impactful on their own. It removes 2 component instances per row. My understanding is that it's not really about the number of elements the component renders, as much as the overhead incurred by having the component instances exist at all: https://vuejs.org/guide/best-practices/performance.html#avoid-unnecessary-component-abstractions - even totally renderless components would incur this cost (per the docs: I haven't measured).

Inlining the SVGs dramatically cuts down on the number of component instances per row, so cumulatively the impact seemed larger for the Highlight Text removals. But my measurement of that is imperfect - it's "how does the scrolling feel to me as I make these changes".

This branch was pretty much a spike to see if anything could be done about Chrome 102, since we have lots of users who will be seeing the UI presented that way. Here's a few thoughts on the debt side:

1. Yes this is debt and less maintainable than using the icon system
2. This in theory could be scheduled to reverted after Chrome 103 is stable and users are using it. Though even Electron 100 was a bit janky for me here, so we might want to keep some of the optimizations. Would need more data/testing to decide.
3. If we really want the virtualized list and will extend the features even further, long term we probably need to be mindful of the number of component instances that are getting mounted in general. We may want to use an SVG sprite setup, not the main icon system, in virtualized lists to keep them as lean as possible - that's probably what a mature version of this spike would look like.
4. We'll hopefully revisit all of this soon anyway for a11y, so there's another opportunity to unwind this.
5. Ultimately, I think we should be filtering and sorting on the server, and possibly doing traditional pagination on the front end, not virtualizing a list of all specs. This would solve, and future-proof, a lot of what's going on here. And then the code would have less need to consider the instances, since rendering time would be less critical no matter what the arch is.